contracts/SpokePool.sol

```diff
@@ -147,19 +179,88 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     }
 
     function initiateRelay(
-        uint256 originChain,
-        address sender,
+        uint256 originChainId,
         uint256 amount,
-        address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
+        uint64 depositId,
+        uint64 relayerFeePct,
+        uint64 realizedLpFeePct,
+        address relayToken,
+        address sender,
+        address recipient
+    ) public {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18 && realizedLpFeePct <= 0.5e18, "invalid fees");
+
+        // Use relay count as unique ID for new relay.
+        require(relays[numberOfRelays].relayAmount == 0, "Relay exists");
+        relays[numberOfRelays] =
+            RelayData(
+                recipient,
+                relayToken,
+                relayerFeePct,
+                realizedLpFeePct,
+                amount, // total relay amount
+                0 // total amount filled
+            );
+            
+        emit InitiatedRelay(
+            originChainId, 
+            amount,
+            depositId, 
+            numberOfRelays,  
+            relayerFeePct, 
+            realizedLpFeePct, 
+            relayToken, 
+            sender, 
+            recipient
+        );
+
+        // Questions:
+        // - do we need to store relayer (i.e. msg.sender) anywhere on-chain? Or can we assume data worker
+        //   can grab it from event data.
+
+        numberOfRelays += 1;
+    }
 
     function fillRelay(
```

Comment:
I like the structure you have here and how you defined `amount` as including fees and then send amount without fees.